tests/unit/downloader/test_entity.py

```diff
@@ -0,0 +1,262 @@
+import uuid
+
+import pytest
+from assertpy import assert_that
+
+from hca_ingest.downloader.schema_url import SchemaUrl
+from hca_ingest.downloader.entity import Entity
+
+
+@pytest.fixture
+def domain_type():
+    return 'protocol'
+
+
+@pytest.fixture
+def concrete_type():
+    return 'sequencing_protocol'
+
+@pytest.fixture
+def schema(domain_type, concrete_type):
+    return SchemaUrl(f'https://schema.humancellatlas.org/type/{domain_type}/sequencing/10.1.0/{concrete_type}')
+
+
+@pytest.fixture
+def metadata_uuid():
+    return str(uuid.uuid4())
+
+
+@pytest.fixture
+def metadata_id() -> str:
+    return str(uuid.uuid4()).replace('-', '')
+
+
+@pytest.fixture
+def populated_entity(schema, metadata_uuid, metadata_id):
+    return Entity({
+        'content': {
+            'describedBy': schema.url
+        },
+        'uuid': {'uuid': metadata_uuid},
+        "_links" : {
+            "self" : {
+                "href": f"https://api.ingest.archive.data.humancellatlas.org/projects/{metadata_id}"
+            }
+        }
+    })
+
+
+def test_populated_entity(populated_entity, metadata_uuid, metadata_id, schema):
+    assert_that(populated_entity.uuid).is_equal_to(metadata_uuid)
+    assert_that(populated_entity.id).is_equal_to(metadata_id)
+    assert_that(populated_entity.schema).is_equal_to(schema)
+
+
+@pytest.fixture
+def schema_empty_string():
+    return {
+        'content': {
+            'describedBy': ''
+        }
+    }
+
+@pytest.fixture
+def schema_none():
+    return {
+        'content': {
+            'describedBy': None
+        }
+    }
+
+@pytest.fixture
+def schema_missing():
+    return {
+        'content': {}
+    }
+
+
+@pytest.fixture
+def content_none():
+    return {
+        'content': None
+    }
+
+
+@pytest.fixture
+def content_missing():
+    return {}
+
+
+@pytest.fixture(params=[
+    pytest.lazy_fixture('schema_empty_string'),
+    pytest.lazy_fixture('schema_none'),
+    pytest.lazy_fixture('schema_missing'),
+    pytest.lazy_fixture('content_none'),
+    pytest.lazy_fixture('content_missing')
+])
+def missing_schema_entity(request):
+    return Entity(request.param)
+
+
+@pytest.fixture
+def missing_schema(missing_schema_entity: Entity) -> SchemaUrl:
+    return missing_schema_entity.schema
+
+
+@pytest.fixture
+def blank_schema() -> SchemaUrl:
+    return SchemaUrl()
+
+
+def test_missing_schema(missing_schema: SchemaUrl, blank_schema: SchemaUrl):
+    assert_that(missing_schema.url).is_equal_to(blank_schema.url)
+    assert_that(missing_schema.concrete_type).is_equal_to(blank_schema.concrete_type)
+    assert_that(missing_schema.domain_type).is_equal_to(blank_schema.domain_type)
```

Comment:
Have you considered just comparing the objects?
```suggestion
    assert_that(missing_schema).is_equal_to(blank_schema)
```

Reply:
this was my first try, but doesn't work, I assume because it's looking for reference equality.

Reply:
scratch that, there must have been something else going on at the time.
Reverting back to the recommended version.